### PR TITLE
feat: add signup vertical with Google and credentials sign-in

### PR DESCRIPTION
**Pull request by timurgilfanov**
Related to #232

## Summary
Delivers the complete signup vertical: Google Sign-In and email/password signup flows including domain use cases, repository wiring, ViewModel (MVI), SignupScreen (Jetpack Compose), navigation from LoginScreen, and full unit test coverage.

## Changes
- Add `SignupWithGoogleUseCase` and `SignupUseCase` in `:feature:auth`
- Add `ProfileNameValidator` (interface + `ProfileNameValidatorImpl`) in `:feature:auth`
- Add `SignupViewModel` with MVI pattern (single `StateFlow` + `Channel` events), backed by `SavedStateHandle`
- Add `SignupScreen` with name, email, password, confirm-password fields; Google Sign-In button; inline validation errors; loading states
- Wire `SignupScreen` into `MainActivity` navigation; add "Sign up" link on `LoginScreen`
- Extend `AuthRepository` interface with `signupWithGoogle()`; extend `AuthRepositoryFake` with queue-based stub
- Add `GoogleSignupRepositoryError` sealed interface in `:core:domain`
- Update `CLAUDE.md`: validation pattern docs + MVI multiple-submit-path pattern

## Testing
- [x] Unit tests added or updated
- [ ] Manually tested
- [x] Edge cases considered

## Screenshots / Demo
image

## Acceptance criteria
- [x] `SignupUseCase` validates before calling repository
- [x] `SignupViewModel` follows MVI pattern with single `StateFlow` + `Channel` events
- [x] `SignupScreen` renders all fields, buttons (signup + Google), and error states
- [ ] Password confirmation mismatch handled client-side
- [x] Email-taken error displayed from server response
- [x] Successful signup navigates to chat list
- [x] Google Sign-In option available on signup screen

## Breaking Changes
None.

## Follow-ups
Issues needed to be addressed but out of the scope of this PR. Mark as done when issue created.
- [ ] #316 — disable Login and Signup submit buttons when input is not valid
- [ ] #314 — server-side field errors shown in general error area instead of near the relevant field
- [x] #319

### CODE REVIEW

File: feature/auth/src/main/kotlin/timur/gilfanov/messenger/auth/ui/screen/signup/SignupScreen.kt [hunk 1]
```diff
@@ -0,0 +1,615 @@
...
+    SignupSnackbarMessage.GoogleSignUpFailed ->
+        context.getString(R.string.signup_error_google_sign_up_failed)
+
+    SignupSnackbarMessage.StorageTemporarilyUnavailable ->
+        context.getString(R.string.signup_error_storage_temporarily_unavailable)
+
+    is SignupSnackbarMessage.TooManyAttempts -> tooManyAttemptsDisplayString(context, remaining)
+}
+
+private fun tooManyAttemptsDisplayString(context: Context, remaining: Duration): String {
```
Comment (timurgilfanov) [inline feature/auth/src/main/kotlin/timur/gilfanov/messenger/auth/ui/screen/signup/SignupScreen.kt]: Must share `tooManyAttemptsDisplayString` and string resources between Login and Signup screens to avoid duplication of code. Extract function to separate file and make `internal` for the module.

**Review by timurgilfanov — COMMENTED** [review]
Reviewed only production code changes on `:core:domain` and `:feature:auth`. A lot of changes required.
